Source/Core/VideoCommon/ShaderGenCommon.h

```diff
@@ -59,27 +52,19 @@ class ShaderGeneratorInterface
 	 */
 	template<class uid_data>
 	uid_data* GetUidData() { return nullptr; }
-
-protected:
-	std::string m_buffer;
 };
 
-/**
+/*
  * Shader UID class used to uniquely identify the ShaderCode output written in the shader generator.
  * uid_data can be any struct of parameters that uniquely identify each shader code output.
  * Unless performance is not an issue, uid_data should be tightly packed to reduce memory footprint.
  * Shader generators will write to specific uid_data fields; ShaderUid methods will only read raw u32 values from a union.
+ * NOTE: Because LinearDiskCache reads and writes the storage associated with a ShaderUid instance, ShaderUid must be a POD struct.
  */
 template<class uid_data>
 class ShaderUid : public ShaderGeneratorInterface
 {
 public:
-	ShaderUid()
-	{
-		// TODO: Move to Shadergen => can be optimized out
-		memset(values, 0, sizeof(values));
```

Comment:
Yeah, I removed the constructor because initially I made the class POD. And as for moving it, pretty much what the comment says, I moved it into shadergen (also added a note in the commit message about it)